src/components/navigation/NavDropDown.tsx

```diff
@@ -0,0 +1,121 @@
+import { SupportedLanguageTypes } from '@/configs'
+import { hulen_black, hulen_yellow, hulen_yellow_text } from '@/styles'
+import { SanityNavElement } from '@/types/sanity'
+import { ArrowDownward, ArrowDropDown } from '@mui/icons-material'
+import { Box, Button, ClickAwayListener, List, ListItem, Typography } from '@mui/material'
+import { usePathname } from 'next/navigation'
+import { MouseEventHandler, useRef, useState, useEffect } from 'react'
+import { DrawerLinkItem } from './DrawerLinkItem'
+
+/** Navigation Dropdown component used in navigation and on mobile devices in MenuDrawer
+ *
+ */
+export const NavDropDown = ({
+  navElement,
+  language,
+  isMobile = false,
+  onClick,
+}: {
+  navElement: SanityNavElement
+  language: SupportedLanguageTypes
+  isMobile?: boolean
+  onClick?: MouseEventHandler<HTMLAnchorElement>
+}) => {
+  const dropdownRef = useRef<HTMLUListElement>(null)
+  const currentPath = usePathname()
+  const [open, setOpen] = useState(false)
+
+  const isCurrentParentPath = currentPath.startsWith(navElement.subUrl)
+
+  const toggleOpen = () => {
+    setOpen((prev) => !prev)
+  }
+
+  //ids for HTML elements used in aria-tags
+  const buttonId = navElement.title[language].replace(' ', '').toLowerCase() + '-btn'
+  const menuId = navElement.title[language].replace(' ', '').toLowerCase() + '-menu'
+
+  //Close dropdown if escape key is pressed
+  useEffect(() => {
+    const close = (e: KeyboardEvent) => {
+      if (e.key == 'Escape') {
+        setOpen(false)
+      }
+    }
+    if (open) {
+      window.addEventListener('keydown', close)
+    }
+
+    return () => window.removeEventListener('keydown', close)
+  }, [open])
+
+  return (
+    <ClickAwayListener onClickAway={() => setOpen(false)}>
+      <Box sx={{ position: 'relative' }}>
+        <Button
+          id={buttonId}
+          sx={{
```

Comment:
Såpass mye attributter i sx at her kan vi heller lage en variant  for button og definere dette i theme